src/xccl/ProcessGroupXCCL.cpp

```diff
@@ -905,6 +910,8 @@ c10::intrusive_ptr<Work> ProcessGroupXCCL::pointToPoint(
         options_->timeout,
         pgStatus_,
         true);
+    work->trace_id_ = traceId.id;
```

Comment:
Same as above. Could you please add a check to see if traceId is of type uint64_t? If it is, assign trace_id_ directly to r->trace_id_; otherwise, parse it using traceId.id.